### DISCUSSION TITLE
Middleware plugin to apply response headers set before forwarding request

### DISCUSSION THREAD

**Discussion by janko** [jeremyevans/roda#258] [General]
This is a follow-up to an [older discussion](https://groups.google.com/g/rodauth/c/H6og6eOYYAQ/m/WiQ9Fas5CQAJ) on the Rodauth google group. Currently, when using a Roda app as middleware, any response headers set in the `route` block are discarded if the request is forwarded to the main app. For Rodauth, this affects extending the remember period, and also possible `Authorization` header updates when using JWT. In rodauth-rails, I had [wrapped the middleware plugin](https://github.com/janko/rodauth-rails/blob/997b3f64a5e02a4ef995e2d57d3554ba151528a2/lib/rodauth/rails/app/middleware.rb) in order to apply the changed response headers, but I consider it a bit of a hack, so I wanted to check if there is interest in an official implementation in the middleware plugin.

I was imagining an additional boolean configuration option for the plugin, where the user can choose whether they want to enable this behaviour. I was thinking the response headers could be stored in Rack env (e.g. `roda.headers`) before throwing `:next` in `_roda_new_main_route`. Then after the main app is called (but before `:handle_result` is called), the saved headers would be retrieved from env and merged into the headers of the main app's response. I was thinking of excluding Roda's default headers here, which I believe wouldn't be set on `response.headers` anyway, as neither `#finish` nor `#finish_with_body` was called before a request is forwarded.

If this sounds reasonable, I could work on a PR for that.

**Comment by jeremyevans** [comment]
I agree with adding the feature, and your implementation approach sounds good.  Please do submit a PR for it when you have time.
